### ISSUE TITLE
Missing impelementation of MacCtx.RadioEvents.IrqProcessNeeded in LoRaMAC.c

### ISSUE THREAD

**Issue by dgiovanelli** [Lora-net/LoRaMac-node#1216]
The callback IrqProcessNeeded is not implemented into LoRaMAC.c. The IRQ generated by the chipset (at least for the LR1110) are therefore processed only if the LmHandlerProcess() function is continuosly polled.

Also, in LoRaMac-node/src/radio/lr1110/radio.c the IrqFired flag is reset at the end of RadioInit( ). However, in my case (I'm using a proprietary hardware), the interrupt line is rised by the chipset before reaching the end of RadioInit( ). This causes the level of the IrqFired flag to go out of sync with the physical interrupt line (the IrqFired flag is cleared while the interrupt line is high).

**Comment by mluis1** [comment]
Yes, on provided examples one need to call `LmHandlerProcess` api in order for the `Radio.IrqProcessed` to be executed.
If your code is not using the LmHandler packages then you must call `Radio.IrqProcessed` before calling `LoRaMacProcess`

Concerning your 2nd question it is hard to provide you an answer. 
My advice would be to check and understand why an interrupt is being generated by your proprietary hardware. The execution of `RadioInit` function should not as far as I know trig an interrupt from the radio.
